src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/CollectionsMarshalTests.cs

```diff
@@ -0,0 +1,47 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+
+namespace System.Runtime.InteropServices.Tests
+{
+    public class CollectionsMarshalTests
```

Comment:
Add a test where `List<T>` is changed after call to `AsSpan` and assert that change isn't reflected in the span.

Reply:
Not sure I understand, altering the contents of the Span will alter the contents of the List and vice versa; unless the link was broken by the List resizing its array; which seems both a fragile thing to test and an undesirable behaviour to enforce via a test? Its an unfortunate constraint of the implementation.

Reply:
> altering the contents of the Span will alter the contents of the List and vice versa

Oh I see. I misunderstood the discussion from the API review.

> unless the link was broken by the List resizing its array; which seems both a fragile thing to test and an undesirable behaviour to enforce via a test? Its an unfortunate constraint of the implementation.

Right, that's what I was thinking off. Effectively, this API is unsafe. Why? Because changes to the list could break the link that the span points to. A test to show-case why this API is unsafe helps as a sample. I don't see how such a test would be fragile though. Isn't it always deterministic? The way I see it is that tests codify the behavior of the API in various scenarios (whether invalid or valid). The fact that it shouldn't be done/leads to incorrect behavior due to the constraint of the API is covered by documentation on how to properly use it (and where we placed the API).

Reply:
> I don't see how such a test would be fragile though. Isn't it always deterministic?

Not without knowledge of the implementation. However, I could just keep adding until `.Capacity` changes and check after the change.